src/node.rs

```diff
@@ -114,7 +133,7 @@ impl Node {
                      compute_current_candidate_blocks(potentially_current));
         self.current_blocks = compute_current_blocks(&self.current_candidate_blocks);
         if is_ambiguous(&self.current_blocks, self.our_name) {
-            self.ambiguous_step = self.ambiguous_step.or(Some(step));
+            self.ambiguous_step = self.ambiguous_step.or_else(|| Some(step));
```

Comment:
Yes, it's this [clippy lint](https://github.com/Manishearth/rust-clippy/wiki#or_fun_call).  I think it does make sense, but is hardly worth bothering about given the cost of the allocation here and that this is a simulator.